rules/terraform_required_version.go

```diff
@@ -78,13 +81,57 @@ func (r *TerraformRequiredVersionRule) Check(runner tflint.Runner) error {
 		exists = exists || ok
 	}
 
-	if !exists {
-		return runner.EmitIssue(
-			r,
-			`terraform "required_version" attribute is required`,
-			hcl.Range{},
-		)
+	if exists {
+		return nil
+	}
+
+	var missingRange hcl.Range
+
+	if len(body.Blocks) > 0 {
+		missingRange = body.Blocks[0].DefRange
+	} else {
+		// If there are no "terraform" blocks, create a hcl.Range for the files
+
+		// Find the directory common to all files
+		var commonPath []string
```

Comment:
Per L49 this rule only runs in root modules. So it should be sufficient to just take the module path and not do all this work around inferring from the list of files, no?

Reply:
That sounds reasonable to me, but I'm admittedly new to terraform and definitely new to this code base, so I'm not sure what assumptions are safe to make.

It was also unclear to me a direct way to get the module path. `runner.Ctx.ModulePath` looked promising, but I wasn't sure how to handle a `addrs.ModuleInstance`.

If it's straight forward to do then, yeah, I think this could be simplified a lot too to something like:

```
...
} else {
    if len(files) == 1 {
        for filename := range files {
            missingRange = hcl.Range{ Filename: filename }
        }
    } else {
        missingRange = hcl.Range{ Filename: < module path > }
    }
}
```

Reply:
Hmm I guess there's no official interface for getting the path on disk, just the path in Terraform (`module.*`). That said, each runner should only operate in a single module (directory). So I think it should be enough to just do `filepath.Dir(files[0])`?

Reply:
Sounds great to me. I've updated it, so I pull a single file from the map. If it's the only file, use it for the `hcl.Range` otherwise if there are multiple files, use `filepath.Dir` to reference the directory of the module.